Empty examples table is not handled properly in composite step

**STR:**
Scenario:
```
When I execute composite step:
```

Composite:
```
Composite: When I execute composite step:$table
When I initialize SCENARIO variable `table` with values:<table>
```

**Expected:**
Variable `table` is created and can be handled further, e.g.:
```
When I initialize the SCENARIO variable `varName` with value `${table[0].someParam:}`
```

**Actual:**
```
org.jbehave.core.failures.UUIDExceptionWrapper: When I initialize SCENARIO variable `table` with values:<table>
        at org.jbehave.core.steps.StepCreator$ParametrisedStep.perform(StepCreator.java:856) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:392) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:399) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$PerformableSteps.perform(PerformableTree.java:1305) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1188) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.performRestartableSteps(PerformableTree.java:1167) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.performScenario(PerformableTree.java:1179) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$NormalPerformableScenario.perform(PerformableTree.java:1219) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:1100) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:987) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:955) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:471) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:438) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:293) [jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:266) [jbehave-core-4.6.4-alpha.3.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:830) [?:?]
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
        at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
        at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[?:?]
        at java.util.Objects.checkIndex(Objects.java:373) ~[?:?]
        at java.util.ArrayList.get(ArrayList.java:425) ~[?:?]
        at org.jbehave.core.model.ExamplesTable.withRows(ExamplesTable.java:253) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        at org.vividus.bdd.steps.ParameterConvertersDecorator.resolvePlaceholdersAfterConversion(ParameterConvertersDecorator.java:93) ~[vividus-bdd-engine-0.2.5-SNAPSHOT.jar:0.2.5-SNAPSHOT]
        at org.vividus.bdd.steps.ParameterConvertersDecorator.convert(ParameterConvertersDecorator.java:78) ~[vividus-bdd-engine-0.2.5-SNAPSHOT.jar:0.2.5-SNAPSHOT]
        at org.jbehave.core.steps.StepCreator.convertParameterValues(StepCreator.java:439) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.steps.StepCreator.access$1400(StepCreator.java:49) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.steps.StepCreator$ParametrisedStep.parametriseStep(StepCreator.java:889) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        at org.jbehave.core.steps.StepCreator$ParametrisedStep.perform(StepCreator.java:830) ~[jbehave-core-4.6.4-alpha.3.jar:?]
        ... 18 more
```

--- a/vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ParameterConvertersDecorator.java
+++ b/vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ParameterConvertersDecorator.java
@@ -84,13 +84,16 @@ public class ParameterConvertersDecorator extends ParameterConverters
         if (type == ExamplesTable.class)
         {
             ExamplesTable examplesTable = (ExamplesTable) result;
-            List<Map<String, String>> rows = examplesTable.getRows();
-            rows.forEach(
-                row -> row.entrySet().forEach(
-                    cell -> cell.setValue((String) resolvePlaceholders(cell.getValue(), String.class))
-                )
-            );
-            examplesTable.withRows(rows);
+            if (!examplesTable.isEmpty())
+            {
+                List<Map<String, String>> rows = examplesTable.getRows();
+                rows.forEach(
+                    row -> row.entrySet().forEach(
+                        cell -> cell.setValue((String) resolvePlaceholders(cell.getValue(), String.class))
+                    )
+                );
+                examplesTable.withRows(rows);
+            }
         }
     }
 

--- a/vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/ParameterConvertersDecoratorTests.java
+++ b/vividus-bdd-engine/src/test/java/org/vividus/bdd/steps/ParameterConvertersDecoratorTests.java
@@ -19,11 +19,13 @@ package org.vividus.bdd.steps;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
 import java.lang.reflect.Type;
@@ -263,5 +265,20 @@ class ParameterConvertersDecoratorTests
         assertThat(result, instanceOf(ExamplesTable.class));
         ExamplesTable table = (ExamplesTable) result;
         assertEquals(List.of(Map.of(expressionKey, expressionValue, variableKey, variableValue)), table.getRows());
+    }
+
+    @Test
+    void shouldConvertToEmptyExamplesTable()
+    {
+        String pathToTable = "/empty-example-table.table";
+        String tableAsString = "";
+        when(expressionAdaptor.process(pathToTable)).thenReturn(pathToTable);
+        when(parameterAdaptor.convert(pathToTable)).thenReturn(pathToTable);
+        when(storyLoader.loadResourceAsText(pathToTable)).thenReturn(tableAsString);
+        Object result = parameterConverters.convert(pathToTable, ExamplesTable.class);
+        assertThat(result, instanceOf(ExamplesTable.class));
+        ExamplesTable table = (ExamplesTable) result;
+        assertTrue(table.isEmpty());
+        verifyNoMoreInteractions(expressionAdaptor, parameterAdaptor);
     }
 }

--- a/vividus-tests/src/main/resources/story/integration/VariableTests.story
+++ b/vividus-tests/src/main/resources/story/integration/VariableTests.story
@@ -19,6 +19,8 @@ When I initialize SCENARIO variable `testMap` with values:
 |value1|value2|
 Then `value1` is = `${testMap[0].key1}`
 Then `value2` is = `${testMap[0].key2}`
+When I initialize SCENARIO variable `testEmptyMap` with values:
+Then `` is = `${testEmptyMap[0].key:}`
 
 Scenario: Verify ability to use variables with names containing special characters
 !-- The following checks verify handling of non-initialized BDD variables

